Apps/APN/src/screens/Dashboard.js

```diff
@@ -150,6 +161,32 @@ const Dashboard = ({ navigation }) => {
     }
   };
 
+  // Setup 3rd party SDK, react-native-notifications
+  // We install this SDK into sample app to make sure the CIO SDK behaves as expected when there is another SDK installed that handles push notifications.
+  //
+  // Important to test that 3rd party SDK is able to decide if a push is shown or not while app is in foreground for non-CIO sent pushes.
+  Notifications.events().registerNotificationReceivedForeground(
+    (notification: Notification, completion) => {
+      console.log(
+        `Non-Customer.io notification received in foreground: ${notification.title} : ${notification.body}`
+      );
+
+      completion({ alert: true, sound: true, badge: true });
+    }
+  );
+  // Important to test that 3rd party SDK is able to receive a callback when a push notification is clicked for non-CIO sent pushes.
+  Notifications.events().registerNotificationOpened(
```

Comment:
>for non-CIO sent

how does it differentiate between CIO and non-CIO?

Reply:
I think that you're asking how does the function registered to `Notifications.events().registerNotificationOpened(` know the difference between CIO and non-CIO pushes? 

If that is true, the function registered to `Notifications.events().registerNotificationOpened(` does not need to know the difference. 

The app's flow when a push is clicked:
* Push clicked
* Native CIO iOS SDK handles push click. 
* CIO SDK determines if the push came from CIO. 
* If the push did not come from CIO, [the native CIO SDK will forward the push click handling to other listeners in the app](https://github.com/customerio/customerio-ios/blob/a316244510039d68a3a17250592238fe16bb8dc9/Sources/MessagingPush/PushHandling/iOSPushEventListener.swift#L144-L147). It's at this point where `Notifications.events().registerNotificationOpened(` function will get called. 

If the push *is* sent from CIO, the function registered to `Notifications.events().registerNotificationOpened(` will *not* get called.

Reply:
Yes, that was exactly my question. Apologies for not asking this earlier in the initial commit, but this use case didn't come in my mind then considering native but for wrapper SDK its a different story. 

But is there any specific reason, we don't want users to get a callback if it's a CIO push? I can see a lot of value in giving them a callback. 

- Customers want to manually track metric 
- Customers have sent custom data, that they want to perform some action to. 

Because these are our actual recommendations to the wrapper customers coming with the following requests? we ask them if they can use any 3rd party library to intercept our push and get the custom data they want to send.

Reply:
The CIO SDK not forwarding the push click event is part of this feature of automatic push click handling. Part of the responsibility of handling a push event is by calling `completionHandler()` when iOS calls `func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void)`. 

This automatic push click handling feature is designed to provide an easy happy path for customers who do not want to manually track metrics. 

Both of the use cases you mentioned: 
```
    Customers want to manually track metric
    Customers have sent custom data, that they want to perform some action to.
```

I would consider to be use cases that can only be done if a customer handles the push click event themselves. Therefore, the customer should disable this automatic push click handling feature and handle the push click themselves.

Reply:
Thanks, Levi for the explanation, so can you just confirm these two things to make sure I understand correctly, it's been quite some time since I looked into the implementation. 

- With this technique used in this feature, It's not technically possible to `track metric` for CIO push and then pass it on to just like we do for non-CIO push? because of the `completionHandler()` ?
- Secondly, wrapper SDK have the config to disable automatic push click?

Reply:
> It's not technically possible to track metric for CIO push and then pass it on to just like we do for non-CIO push? because of the completionHandler() ?

I am not sure I would agree with "it's not technically possible" but instead answer this question by saying, the automatic push handling feature would not be fully automatic if we still required customers to call the `completionHandler` in their push click handlers. The code that handles a push (customer's app, our SDK, a 3rd party SDK) is supposed to call the `completionHandler` callback method if that code handled the push. This feature is meant to handle CIO pushes for you with no need to add code yourself. So if this feature is enabled, it's our responsibility to handle the push and call `completionHandler`. 

> Secondly, wrapper SDK have the config to disable automatic push click?

All SDKs (native or wrapper) can disable this feature by configuring the SDK inside of the `AppDelegate`. 

```swift
CustomerIO.initialize(...) { config in
  config.autoTrackPushEvents = false 
}
```

Reply:
Awesome, appreciate the clarity on things. I believe we still need to add an attribute in `JS/Dart` that would then map to `config.autoTrackPushEvents = false `

Because customers don't call, 
```
CustomerIO.initialize(...) { config in
  config.autoTrackPushEvents = false 
}
``` 
in their App, we call it from our SDK from the config customers have provided us.

Reply:
Talked in DM! we are good on this.